source/credentials_provider_profile.c

```diff
@@ -374,13 +384,30 @@ struct aws_credentials_provider *aws_credentials_provider_new_profile(
     struct aws_string *credentials_file_path = NULL;
     struct aws_string *config_file_path = NULL;
     struct aws_string *profile_name = NULL;
+    bool first_profile_in_chain = false;
+    if (source_profiles_table == NULL) {
+        source_profiles_table = aws_mem_calloc(allocator, 1, sizeof(struct aws_hash_table));
+        first_profile_in_chain = true;
+        if (aws_hash_table_init(
+                source_profiles_table,
+                allocator,
+                3,
+                aws_hash_byte_cursor_ptr,
+                (aws_hash_callback_eq_fn *)aws_byte_cursor_eq,
```

Comment:
any function type casting is ub. it will probably work in most cases, but its still better to wrap